wgpu-core/src/validation.rs

```diff
@@ -170,6 +170,70 @@ fn get_aligned_type_size(
     }
 }
 
+fn map_storage_format_to_naga(format: wgt::TextureFormat) -> Option<naga::StorageFormat> {
+    use naga::StorageFormat as Sf;
+    use wgt::TextureFormat as Tf;
+    // Using the table in https://gpuweb.github.io/gpuweb/#plain-color-formats
+    Some(match format {
+        Tf::R32Uint => Sf::R32Uint,
+        Tf::R32Sint => Sf::R32Sint,
+        Tf::R32Float => Sf::R32Float,
+        Tf::Rgba8Unorm => Sf::Rgba8Unorm,
+        Tf::Rgba8Snorm => Sf::Rgba8Snorm,
+        Tf::Rgba8Uint => Sf::Rgba8Uint,
+        Tf::Rgba8Sint => Sf::Rgba8Sint,
+        Tf::Rg32Uint => Sf::Rg32Uint,
+        Tf::Rg32Sint => Sf::Rg32Sint,
+        Tf::Rg32Float => Sf::Rg32Float,
+        Tf::Rgba16Uint => Sf::Rgba16Uint,
+        Tf::Rgba16Sint => Sf::Rgba16Sint,
+        Tf::Rgba16Float => Sf::Rgba16Float,
+        Tf::Rgba32Uint => Sf::Rgba32Uint,
+        Tf::Rgba32Sint => Sf::Rgba32Sint,
+        Tf::Rgba32Float => Sf::Rgba32Float,
+        _ => return None,
+    })
+}
+
+fn map_storage_format_from_naga(format: naga::StorageFormat) -> wgt::TextureFormat {
+    use naga::StorageFormat as Sf;
+    use wgt::TextureFormat as Tf;
+    match format {
+        Sf::R8Unorm => Tf::R8Unorm,
+        Sf::R8Snorm => Tf::R8Snorm,
+        Sf::R8Uint => Tf::R8Uint,
+        Sf::R8Sint => Tf::R8Sint,
+        Sf::R16Uint => Tf::R16Uint,
+        Sf::R16Sint => Tf::R16Sint,
+        Sf::R16Float => Tf::R16Float,
+        Sf::Rg8Unorm => Tf::Rg8Unorm,
+        Sf::Rg8Snorm => Tf::Rg8Snorm,
+        Sf::Rg8Uint => Tf::Rg8Uint,
+        Sf::Rg8Sint => Tf::Rg8Sint,
+        Sf::R32Uint => Tf::R32Uint,
+        Sf::R32Sint => Tf::R32Sint,
+        Sf::R32Float => Tf::R32Float,
+        Sf::Rg16Uint => Tf::Rg16Uint,
+        Sf::Rg16Sint => Tf::Rg16Sint,
+        Sf::Rg16Float => Tf::Rg16Float,
+        Sf::Rgba8Unorm => Tf::Rgba8Unorm,
+        Sf::Rgba8Snorm => Tf::Rgba8Snorm,
+        Sf::Rgba8Uint => Tf::Rgba8Uint,
+        Sf::Rgba8Sint => Tf::Rgba8Sint,
+        Sf::Rgb10a2Unorm => Tf::Rgb10a2Unorm,
+        Sf::Rg11b10Float => Tf::Rg11b10Float,
```

Comment:
A couple of these formats appear to be missing from the inverse transform. Is this intentional?

Reply:
This is a little bit tricky. So right now, the set of formats supported for storage is smaller than what Naga has.
We could reduce Naga's set, but then it would stop being useful for non-WebGPU-related things, e.g. for gfx-rs, where more formats can be storage.

`map_storage_format_to_naga` only has this limited subset.
In `map_storage_format_from_naga` I decided to map all the formats, so that it's 100% succeeds. Then, if see that an inappropriate format is used, I can generate a `BadStorageFormat` error. If I had the `map_storage_format_from_naga` returning an `Option`/`Result`, I'd have to figure out a different error variant for it...

Reply:
Ah alright; seems reasonable.